### DISCUSSION TITLE
How to repeat/broadcast an nnx.Param type?

### DISCUSSION THREAD

**Discussion by ojss** [google/flax#4559] [Q&A]
In torch you can repeat/broadcast an `nn.Param` and use it for further processing. This is useful for trainable initialization states of LSTMs and latent variable models. And even in the case of masked auto encoders (although no repetition occurs there).
How can I repeat a `nnx.Param` type using einops or jnp#tile or jnp#repeat before using it in `nnx.Scan`?

**Comment by ojss** [comment]
bumping this up. @cgarciae perhaps you have some insights?

**Reply by ojss** [reply]
Would really appreciate any available insight into this!

**Reply by DiagRisker** [reply]
Hello ojss, maybe a minimal example would help ! in torch for the part you want to translate and part code in nnx for what you are sure

**Reply by ojss** [reply]
Hi thanks for responding!

Maybe something like this below:

```python
p = nn.Parameter(torch.randn((1, 512))
p = einops.repeat(p, "1 d -> b d", b=64)
```

**Reply by DiagRisker** [reply]
Hello , sorry for the time taken, this question is more of a JAX matter , suppose you have 
```python
p = jnp.ones( (1,512))
p = jnp.concatenate( [p]*64, axis=0)
``` 
Now for a broader context, I'm not sure if you want to do repeat for more dimensions. If yes, you can do:
```python
p = jnp.ones( (1,512))
p = jnp.concatenate( [p]*64*3, axis=0).reshape(64,3, len(p))
```
Does this meet your expectation?
